src/NzbDrone.Core/Download/Clients/qBittorrent/QBittorrentProxy.cs

```diff
@@ -0,0 +1,210 @@
+        }
+
+        public void SetTorrentLabel(string hash, string label, QBittorrentSettings settings)
+        {
+            var request = new RestRequest("/command/setLabel", Method.POST);
+            request.AddParameter("hashes", hash);
+            request.AddParameter("label", label);
+
+            var client = BuildClient(settings);
+            var response = ProcessRequest(client, request, settings);
+            response.ValidateResponse(client);
+        }
+
+        public void MoveTorrentToTopInQueue(string hash, QBittorrentSettings settings)
+        {
+            var request = new RestRequest("/command/topPrio", Method.POST);
+            request.AddParameter("hashes", hash);
+
+            var client = BuildClient(settings);
+            var response = ProcessRequest(client, request, settings);
+
+            // qBittorrent rejects all Prio commands with 403: Forbidden if Options -> BitTorrent -> Torrent Queueing is not enabled
+            if (response.StatusCode == HttpStatusCode.Forbidden)
+            {
+                return;
+            }
+
+            response.ValidateResponse(client);
+        }
+
+        private IRestResponse ProcessRequest(IRestClient client, IRestRequest request, QBittorrentSettings settings)
+        {
+            var response = client.Execute(request);
+
+            if (response.ResponseStatus != ResponseStatus.Completed)
+            {
+                throw new DownloadClientException("Transport error: {0}", response.ResponseStatus);
+            }
+
+            if (response.StatusCode == HttpStatusCode.Forbidden)
+            {
+                _logger.Info("Authentication required, logging in.");
+
+                var logged_in = _logins.Get(settings.Username + settings.Password, () => Login(client, settings), _loginTimeout);
+
+                if (!logged_in)
+                {
+                    throw new DownloadClientAuthenticationException("Failed to authenticate");
+                }
+
+                // success! retry the original request
+                response = client.Execute(request);
+            }
+
+            return response;
+        }
+
+        private bool Login(IRestClient client, QBittorrentSettings settings)
+        {
+            var request = new RestRequest("/login", Method.POST);
+            request.AddParameter("username", settings.Username);
+            request.AddParameter("password", settings.Password);
+
+            var response = client.Execute(request);
+
+            if (response.ResponseStatus != ResponseStatus.Completed)
+            {
+                throw new DownloadClientException("Transport error: {0}", response.ResponseStatus);
+            }
+
+            if (response.StatusCode != HttpStatusCode.OK)
+            {
+                _logger.Warn("Login failed with {0}.", response.StatusCode);
+                return false;
+            }
+
+            if (response.Content != "Ok.") // returns "Fails." on bad login
```

Comment:
No, it's either 'Ok.' or 'Fails.' according to https://github.com/qbittorrent/qBittorrent/blob/master/src/webui/webapplication.cpp#L180